// react-solidgauge/dist/Chart/SolidGauge.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Chart = require('react-offcharts-core/Components/Chart');

var _Chart2 = _interopRequireDefault(_Chart);

var _ReactIf = require('react-offcharts-core/Components/ReactIf');

var _ReactIf2 = _interopRequireDefault(_ReactIf);

var _Shadow = require('react-offcharts-core/Components/Defs/Shadow');

var _Shadow2 = _interopRequireDefault(_Shadow);

var _guid = require('react-offcharts-core/Utils/guid');

var _guid2 = _interopRequireDefault(_guid);

var _Group = require('./Group');

var _Group2 = _interopRequireDefault(_Group);

var _PathGroup = require('./PathGroup');

var _PathGroup2 = _interopRequireDefault(_PathGroup);

var _MouseContainerOverlay = require('./MouseContainerOverlay');

var _MouseContainerOverlay2 = _interopRequireDefault(_MouseContainerOverlay);

var _props = require('../Helpers/props');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var isEmpty = function isEmpty(obj) {
  return Object.keys(obj).length === 0 && obj.constructor === Object;
};

var SolidGauge = function SolidGauge(props) {
  return _react2.default.createElement(
    _Chart2.default,
    {
      responsive: props.responsive,
      width: props.width,
      height: props.height
    },
    _react2.default.createElement(
      _Group2.default,
      {
        pathWidth: props.pathWidth,
        pathMargin: props.pathMargin,
        chartMargin: props.chartMargin,
        values: props.values,
        endAngle: props.endAngle
      },
      _react2.default.createElement(_PathGroup2.default, {
        background: props.background,
        fontSize: props.fontSize,
        enterAnimation: props.enterAnimation,
        animationEase: props.animationEase,
        animationTime: props.animationTime,
        circleRadius: props.circleRadius,
        endAngle: props.endAngle,
        shadow: !isEmpty(props.shadow)
      }),
      _react2.default.createElement(
        _ReactIf2.default,
        {
          el: _react2.default.createElement('g', null),
          copy: true,
          condition: props.showTooltip
        },
        _react2.default.createElement(_MouseContainerOverlay2.default, {
          mouseFill: props.mouseFill,
          mouseStrokeWidth: props.mouseStrokeWidth,
          selectable: props.selectable
        })
      ),
      _react2.default.createElement(
        _ReactIf2.default,
        { el: _react2.default.createElement('g', null), condition: !isEmpty(props.shadow) },
        _react2.default.createElement(_Shadow2.default, {
          id: 'shadows',
          x: props.shadow.x,
          y: props.shadow.y,
          width: props.shadow.width,
          height: props.shadow.height,
          dx: props.shadow.dx,
          dy: props.shadow.dy,
          stdDeviation: props.shadow.stdDeviation
        })
      )
    )
  );
};

SolidGauge.propTypes = {
  enterAnimation: _propTypes2.default.bool,
  responsive: _propTypes2.default.bool,
  width: _propTypes2.default.number,
  height: _propTypes2.default.number,
  values: _props.dataShape,
  pathWidth: _propTypes2.default.number,
  pathMargin: _propTypes2.default.number,
  endAngle: _propTypes2.default.number,
  background: _props.fillAndStroke,
  fontSize: _propTypes2.default.number,
  chartMargin: _propTypes2.default.number,
  showTooltip: _propTypes2.default.bool,
  animationTime: _propTypes2.default.number,
  animationEase: _propTypes2.default.string,
  circleRadius: _propTypes2.default.number,
  mouseFill: _propTypes2.default.string,
  mouseStrokeWidth: _propTypes2.default.number,
  selectable: _propTypes2.default.bool,
  shadow: _propTypes2.default.shape({
    id: _propTypes2.default.string,
    y: _propTypes2.default.string,
    x: _propTypes2.default.string,
    height: _propTypes2.default.string,
    width: _propTypes2.default.string,
    stdDeviation: _propTypes2.default.number,
    dx: _propTypes2.default.number,
    dy: _propTypes2.default.number
  })
};

SolidGauge.defaultProps = {
  chartMargin: 50,
  shadow: {},
  background: {}
};

exports.default = SolidGauge;
