// react-offcharts-core/Components/Defs/Gradients.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RadialGradient = exports.LinearGradient = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _d3Color = require('d3-color');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var createStops = function createStops(fill) {
  var fillShades = (0, _d3Color.color)(fill);
  return [_react2.default.createElement('stop', { key: 1, offset: '0', stopColor: fillShades.darker(0.8) }), _react2.default.createElement('stop', { key: 2, offset: '0.25', stopColor: fillShades.darker(0.4) }), _react2.default.createElement('stop', { key: 3, offset: '0.5', stopColor: fillShades }), _react2.default.createElement('stop', { key: 4, offset: '0.75', stopColor: fillShades.brighter(0.4) }), _react2.default.createElement('stop', { key: 5, offset: '1', stopColor: fillShades.brighter(0.8) })];
};

var LinearGradient = exports.LinearGradient = function LinearGradient(_ref) {
  var id = _ref.id,
      x1 = _ref.x1,
      x2 = _ref.x2,
      y1 = _ref.y1,
      y2 = _ref.y2,
      stops = _ref.stops,
      gradientUnits = _ref.gradientUnits;
  return _react2.default.createElement(
    'defs',
    null,
    _react2.default.createElement(
      'linearGradient',
      {
        id: id,
        x1: x1,
        x2: x2,
        y1: y1,
        y2: y2,
        gradientUnits: gradientUnits
      },
      stops
    )
  );
};

LinearGradient.propTypes = {
  id: _propTypes2.default.string,
  x1: _propTypes2.default.string,
  x2: _propTypes2.default.string,
  y1: _propTypes2.default.string,
  y2: _propTypes2.default.string,
  gradientUnits: _propTypes2.default.string,
  stops: _propTypes2.default.oneOfType([_propTypes2.default.arrayOf(_propTypes2.default.node), _propTypes2.default.node])
};

LinearGradient.defaultProps = {
  x1: 0,
  x2: 0,
  y1: 0,
  y2: 1,
  gradientUnits: 'objectBoundingBox'
};

var RadialGradient = exports.RadialGradient = function RadialGradient(_ref2) {
  var id = _ref2.id,
      cx = _ref2.cx,
      cy = _ref2.cy,
      r = _ref2.r,
      fx = _ref2.fx,
      fy = _ref2.fy,
      stops = _ref2.stops,
      gradientUnits = _ref2.gradientUnits;
  return _react2.default.createElement(
    'defs',
    null,
    _react2.default.createElement(
      'radialGradient',
      {
        id: id,
        cx: cx,
        cy: cy,
        r: r,
        fx: fx,
        fy: fy,
        gradientUnits: gradientUnits
      },
      stops
    )
  );
};

RadialGradient.propTypes = {
  id: _propTypes2.default.string,
  cy: _propTypes2.default.number,
  cx: _propTypes2.default.number,
  r: _propTypes2.default.number,
  fx: _propTypes2.default.number,
  fy: _propTypes2.default.number,
  gradientUnits: _propTypes2.default.string,
  stops: _propTypes2.default.oneOfType([_propTypes2.default.arrayOf(_propTypes2.default.node), _propTypes2.default.node])
};

RadialGradient.defaultProps = {
  gradientUnits: 'objectBoundingBox'
};

// If the user sets gradient by type

exports.default = function (props) {
  var stops = props.stops,
      fill = props.fill;

  var theseStops = stops;
  // if the gradient does not have any children then
  // return, no work to be done
  if (!stops || stops.length === 0) {
    theseStops = createStops(fill);
  }
  if (props.type === 1) {
    return LinearGradient(Object.assign({}, props, { stops: theseStops }));
  }
  return RadialGradient(Object.assign({}, props, { stops: theseStops }));
};
