// react-offcharts-core/Components/Defs/Shadow.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var shadows = function shadows(props) {
  return _react2.default.createElement(
    'defs',
    null,
    _react2.default.createElement(
      'filter',
      {
        id: props.id,
        y: props.y,
        x: props.x,
        height: props.height,
        width: props.width
      },
      _react2.default.createElement('feGaussianBlur', {
        'in': 'SourceAlpha',
        stdDeviation: props.stdDeviation,
        result: 'blur'
      }),
      _react2.default.createElement('feOffset', {
        'in': 'blur',
        dx: props.dx,
        dy: props.dy,
        result: 'offsetBlur'
      }),
      _react2.default.createElement(
        'feMerge',
        null,
        _react2.default.createElement('feMergeNode', { 'in': 'offsetBlur' }),
        _react2.default.createElement('feMergeNode', { 'in': 'SourceGraphic' })
      )
    )
  );
};

shadows.propTypes = {
  id: _propTypes2.default.string,
  y: _propTypes2.default.string,
  x: _propTypes2.default.string,
  height: _propTypes2.default.string,
  width: _propTypes2.default.string,
  stdDeviation: _propTypes2.default.number,
  dx: _propTypes2.default.number,
  dy: _propTypes2.default.number
};

shadows.defaultProps = {
  id: 'shadow',
  y: '-10%',
  x: '-10%',
  height: '130%',
  width: '130%',
  stdDeviation: 2,
  dx: 2,
  dy: 2
};

exports.default = shadows;
